apiserver/facades/client/application/deployrepository.go

```diff
@@ -131,26 +137,55 @@ func (api *DeployFromRepositoryAPI) DeployFromRepository(arg params.DeployFromRe
 		NumUnits:          dt.numUnits,
 		Placement:         dt.placement,
 		Constraints:       dt.constraints,
-		Resources:         dt.resources,
+		Resources:         pendingIDs,
 	})
+
 	if err != nil {
```

Comment:
If AddApplication fails, the pending resources are not removed.

https://github.com/juju/juju/blob/12e8743cb1bac3be0817db438a5b6c1f583a8dc6/apiserver/facades/client/application/application.go#L317-L330